ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/projectimport/wizard/ImportProjectNotificationSubscriberImpl.java

```diff
@@ -13,117 +13,135 @@
+                        Log.error(getClass(), throwable);
+                    }
+                }).catchError(promiseErrorHandler);
             }
         };
 
-        try {
-            messageBus.subscribe(wsChannel, importProjectOutputWShandler);
-        } catch (WebSocketException e1) {
-            Log.error(ImportProjectWizardPresenter.class, e1);
-        }
+        messageBusPromise.then(new Operation<MessageBus>() {
+            @Override
+            public void apply(final MessageBus messageBus) throws OperationException {
+                try {
+                    messageBus.subscribe(wsChannel, subscriptionHandler);
+                } catch (WebSocketException wsEx) {
+                    Log.error(ImportProjectNotificationSubscriberImpl.class, wsEx);
+                }
+            }
+        }).catchError(promiseErrorHandler);
     }
 
     @Override
     public void onSuccess() {
-        try {
-            messageBus.unsubscribe(wsChannel, importProjectOutputWShandler);
-        } catch (WebSocketException e) {
-            Log.error(getClass(), e);
-        }
-        notification.setStatus(Notification.Status.FINISHED);
-        notification.setMessage(locale.importProjectMessageSuccess());
+        messageBusPromise.then(new Operation<MessageBus>() {
+            @Override
+            public void apply(MessageBus messageBus) throws OperationException {
+                try {
+                    messageBus.unsubscribe(wsChannel, subscriptionHandler);
+                } catch (WebSocketException e) {
+                    Log.error(getClass(), e);
+                }
+                notification.setStatus(Notification.Status.FINISHED);
+                notification.setMessage(locale.importProjectMessageSuccess(projectName));
+            }
+        }).catchError(promiseErrorHandler);
     }
 
     @Override
-    public void onFailure(String errorMessage) {
-        try {
-            messageBus.unsubscribe(wsChannel, importProjectOutputWShandler);
-        } catch (WebSocketException e) {
-            Log.error(getClass(), e);
-        }
-        notification.setStatus(Notification.Status.FINISHED);
-        notification.setType(Notification.Type.ERROR);
-        notification.setImportant(true);
-        notification.setMessage(errorMessage);
+    public void onFailure(final String errorMessage) {
+        messageBusPromise.then(new Operation<MessageBus>() {
+            @Override
+            public void apply(MessageBus messageBus) throws OperationException {
+                try {
+                    messageBus.unsubscribe(wsChannel, subscriptionHandler);
+                } catch (WebSocketException e) {
+                    Log.error(getClass(), e);
+                }
+                notification.setStatus(Notification.Status.FINISHED);
+                notification.setType(Notification.Type.ERROR);
```

Comment:
I don't really know the truth but i think that in this case it would be better to set notification type before setting notification status, as `FINISHED` looks like trigger for notification progress stop.